### ISSUE TITLE
Questions about reproducing training results

### ISSUE THREAD

**Issue by yuerbujin** [Ram81/habitat-imitation-baselines#15]
hello @Ram81 , i'm facing the same problems as @me-no-money mentioned here https://github.com/Ram81/habitat-imitation-baselines/issues/13#issuecomment-1518502299.

my training-metrics of objectnav task is as follows. But it appears to be overfitting.

![f53dec9f29c5e7c3a20bf368ad733f20](https://user-images.githubusercontent.com/68276188/235343501-607bd10e-4623-48f0-8baf-0aff3d1921c3.png)

i use the following command.

```
python -u -m habitat_baselines.run \
--exp-config habitat_baselines/config/objectnav/il_objectnav.yaml \
--run-type train \
SIMULATOR_GPU_ID 3 \
TORCH_GPU_ID 4 \
NUM_PROCESSES 5 \
NUM_UPDATES 1000000 \
TENSORBOARD_DIR tb\
```

**Comment by yuerbujin** [comment]
should I modify some parameters in the training config? but i don't know how to do the parameter tuning.

**Comment by me-no-money** [comment]
@Ram81, I am using the habitat-baselines/config/objectnav/il_objectnav.yaml. I trained for 10k updates, and evaluated the last trainning checkpoint on 500 episodes. My training results are shown below:
![train=10k-eval_episode=500-train](https://user-images.githubusercontent.com/71805391/235854450-4eb0bfbe-f877-4e6d-9a29-d533941e1ce8.png)

And the evaluation results of the last checkpoint is as follows:
![updates=10k-eval_episode=500](https://user-images.githubusercontent.com/71805391/235852262-089f0465-f4ff-4658-857d-10f744ac5656.png)
I noticed the annotation at the beginning of habitat-baselines/config/objectnav/il_objectnav.yaml sayed that "# Note:  This is an example config, see habitat_baselines/config/pointnav/ppo_pointnav.yaml for better hyperparameters for actual training". Should I use the ppo_pointnav.yaml？But it seems that the ppo_pointnav.yaml is using the RL-PPO not Behavior Cloning algorithm. Or could I copy the hyperparameters of the ppo_pointnav.yaml to the il_objectnav.yaml then train the objectnav task?

**Comment by me-no-money** [comment]
> @Ram81, I am using the habitat-baselines/config/objectnav/il_objectnav.yaml. I trained for 10k updates, and evaluated the last trainning checkpoint on 500 episodes. My training results are shown below: ![train=10k-eval_episode=500-train](https://user-images.githubusercontent.com/71805391/235854450-4eb0bfbe-f877-4e6d-9a29-d533941e1ce8.png)
> 
> And the evaluation results of the last checkpoint is as follows: ![updates=10k-eval_episode=500](https://user-images.githubusercontent.com/71805391/235852262-089f0465-f4ff-4658-857d-10f744ac5656.png) I noticed @the annotation at the beginning of habitat-baselines/config/objectnav/il_objectnav.yaml sayed that "# Note: This is an example config, see habitat_baselines/config/pointnav/ppo_pointnav.yaml for better hyperparameters for actual training". Should I use the ppo_pointnav.yaml？But it seems that the ppo_pointnav.yaml is using the RL-PPO not Behavior Cloning algorithm. Or could I copy the hyperparameters of the ppo_pointnav.yaml to the il_objectnav.yaml then train the objectnav task?

Hi, @Ram81! I haven't been able to reproduce the data like you did. Could you give me a solution of this issue?
BTW, I can't figure out why the success rate of training went from 0 to 1 in just one update. 
![success数据异常](https://user-images.githubusercontent.com/71805391/236629446-2bd6b525-2d71-4cad-b97b-9a91db1ed6d0.png)
Is this normal? Or do I have something wrong?

**Comment by Ram81** [comment]
Hi @yuerbujin @me-no-money,

The training metrics look fine to me. During IL training we only replay the demonstrations from the buffers so success rates should be ~1.0 as we only save human demonstrations that succeeded. In some cases you'll see zero success rate because some demonstrations are successful but because they took more than 500 steps to complete they were marked failed.

While training what you should be looking at is either the `loss` or `action_loss`. That will be the imitation learning loss during training which is a metric to monitor. To figure out whether your models have converged use evaluation success as the metric to monitor.

**Comment by Ram81** [comment]
Closing this issue due to inactivity
